packages/cohort_banking/src/callbacks/statemap_installer.rs

```diff
@@ -1,68 +1,126 @@
+    }
+}
 
-                let sql = r#"
-                    UPDATE bank_accounts ba SET
-                        "amount" =
-                            (CASE
-                                WHEN ba."number" = ($1)::TEXT THEN ba."amount" + ($3)::DECIMAL
-                                WHEN ba."number" = ($2)::TEXT THEN ba."amount" - ($3)::DECIMAL
-                            END),
-                        "version" = ($4)::BIGINT
-                    WHERE ba."number" IN (($1)::TEXT, ($2)::TEXT)
-                        AND ba."version" < ($4)::BIGINT
-                    "#;
+#[async_trait]
+impl ReplicatorInstaller for BankStatemapInstaller {
+    /// Install statemaps and version for respective rows in the `bank_accounts` table and update the `snapshot_version` in `cohort_snapshot` table.
+    ///
+    /// Certain errors like `could not serialize access due to concurrent update` in postgres are retryable.
+    /// - Updating `bank_account` workflow.
+    ///     - On successful completion, we proceed to update the `snapshot_version` in `cohort_snapshot`.
+    ///     - When there is a retryable error, we wait for `retry_wait_ms` milliseconds and go to the start of the loop to retry. `(Txn aborts and retries)`
+    ///     - If all the retries are exhausted, we return `ReplicatorInstallStatus::Gaveup`. `(Txn aborts and returns)`
+    ///     - For non-retryable error, we return the `Error`.
+    /// - Updating `cohort_snapshot` workflow.
+    ///     - On successful completion, we commit the transaction and return `ReplicatorInstallStatus::Installed`. `(Txn Commits and returns)`
+    ///     - When there is a retryable error, we wait for `retry_wait_ms` milliseconds and go to the start of the loop to retry. `(Txn aborts and retries)`
+    ///     - If all the retries are exhausted, we return `ReplicatorInstallStatus::InstalledWithoutSnapshotUpdate`, denoting a updates for bank_transfer went through but snapshot wasn't updated. `(Txn Commits and returns)`
+    ///     - For non-retryable error, we return the `Error`.
 
-                let params: &[&(dyn ToSql + Sync)] = &[&request.from, &request.to, &request.amount, &(sti.version as i64)];
+    async fn install(&self, statemap: Vec<StatemapItem>, snapshot_version: u64) -> Result<ReplicatorInstallStatus, String> {
+        let mut bank_transfer_error_retry_check = retry_allowed_on_error(self.max_retry);
+        let mut snapshot_retry_check = retry_allowed_on_error(self.max_retry);
 
-                let updated_rows = tx.execute(sql, params).await.map_err(|e| e.to_string())?;
+        let mut cnn = self.database.get().await.map_err(|e| e.to_string())?;
+        loop {
+            let tx = cnn.transaction().await.map_err(|e| e.to_string())?;
+            if !statemap.is_empty() {
+                let updated_rows_res = BankStatemapInstaller::update_bank_transfer_request(&tx, &statemap, snapshot_version).await;
```

Comment:
I suggest to rename this method "update_bank_transfer_request". Be reading it feels like we are updating the request. However we are installing the given state map.

Reply:
will update this.